frontend/packages/console-dynamic-plugin-sdk/src/extensions/dashboards.ts

```diff
@@ -0,0 +1,469 @@
+    /** Loader for popup row component */
+    operatorRowLoader?: EncodedCodeRef;
+    /** Links to all resources page. If not provided then a list page of the first resource from resources prop is used. */
+    viewAllLink?: string;
+  };
+
+  export type DashboardsOverviewHealthOperatorCodeRefs<
+    R extends K8sResourceCommon = K8sResourceCommon
+  > = {
+    resources: CodeRef<FirehoseResource[]>;
+    getOperatorsWithStatuses?: CodeRef<GetOperatorsWithStatuses<R>>;
+    operatorRowLoader?: LazyLoader<OperatorRowProps<R>>;
+  };
+
+  /** Adds an inventory status group */
+  export type DashboardsInventoryItemGroup = {
+    /** The id of the status group. */
+    id: string;
+    /** React component representing status group icon. */
+    icon: EncodedCodeRef;
+  };
+
+  export type DashboardsInventoryItemGroupCodeRefs = {
+    icon: CodeRef<React.ReactElement>;
+  };
+
+  /** Adds a resource tile to the overview inventory card */
+  export type DashboardsOverviewInventoryItem = {
+    /** The model for `resource` which will be fetched. The model is used for getting model's label or abbr. */
+    model: EncodedCodeRef;
+    /** Function which will map various statuses to groups. */
+    mapper?: EncodedCodeRef;
+    /** Additional resources which will be fetched and passed to `mapper` function. */
+    additionalResources?: EncodedCodeRef;
+    /** Defines whether the model's label or abbr should be used when rendering the item. Defaults to false (label). */
+    useAbbr?: boolean;
+    /** Loader for the component which will be used when item is expanded. */
+    expandedComponent?: EncodedCodeRef;
+  };
+
+  export type DashboardsOverviewInventoryItemCodeRefs = {
+    model: CodeRef<K8sKind>;
+    mapper?: CodeRef<StatusGroupMapper>;
+    additionalResources?: CodeRef<WatchK8sResources<any>>;
+    expandedComponent?: CodeRef<LazyLoader<ExpandedComponentProps>>;
+  };
+
+  /** Adds an item to the overview utilization card */
+  export type DashboardsOverviewUtilizationItem = {
+    /** The utilization item to be replaced */
+    id: string;
+    /** The Prometheus utilization query */
+    query: string;
+    /** The Prometheus total query */
+    totalQuery: string;
+  };
+
+  /** Adds an activity to the Activity Card of Overview Dashboard where the triggering of activity  is based on watching a K8s resource. */
+  export type DashboardsOverviewResourceActivity = {
+    /** The utilization item to be replaced */
+    k8sResource: EncodedCodeRef;
+    /**
+     * Function which will determine if given resource represents the action.
+     * If the function is not defined, every resource represents activity.
+     */
+    isActivity?: EncodedCodeRef;
+    /** Timestamp for given action, which will be used for ordering */
+    getTimestamp?: EncodedCodeRef;
+    /** Loader for corresponding action component */
+    loader: EncodedCodeRef;
+  };
+
+  export type DashboardsOverviewResourceActivityCodeRefs = {
```

Comment:
we should add generics here too so we allow more specific type than `K8sResourceKind`